lib/msal-node/test/client/ClientCredentialClient.spec.ts

```diff
@@ -446,6 +446,27 @@ describe("ClientCredentialClient unit tests", () => {
                     CONFIDENTIAL_CLIENT_AUTHENTICATION_RESULT.body.access_token
                 );
                 expect(cachedAuthResult.fromCache).toBe(true);
+
+                // validate that the same request but with new claims will make a STS request
```

Comment:
But will a request **without** claims now result in a cache hit or a cache miss ?

In orther words, the scenario: 

1. Get a token without claims. Expected source: IDP
2. Get a token with claims. Expected source: IDP
3. Get a token with other claims. Expected source: IDP (you tested this I think)
4. Get a token without . Expected source: cache

Reply:
Have a look at https://learn.microsoft.com/en-us/entra/identity-platform/app-resilience-continuous-access-evaluation?tabs=JavaScript for current guidance on CAE

Reply:
I just tested it. What happens in this case is that token #4 is the same token as token #1; #4 is retrieved from the cache. They both have a non-existant claims hash, so they match each other.

This comparison logic already existed, but just wasn't being used properly.

However, as I understand the CAE docs you linked, it appears that an application would never receive the same claims challenge twice because after the first request, a new token would be retrieved from the IDP, and the token would pass CAE. Is that right? In that case, we can do what you originally suggested and just skip the cache if there are claims in the token request.

Reply:
How are the access tokens cached? Is the claims_hash made part of the token cache key?

Let's have a meeting with the other MSAL JS folks to discuss about this before proceeding.

Reply:
```
How are the access tokens cached? Is the claims_hash made part of the token cache key?
```

Please avoid this. Just cleaning up a mess with Teams as we added this for them and it is causing cache saturation issues.